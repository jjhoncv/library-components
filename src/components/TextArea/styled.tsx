import styled from 'styled-components';
import { COLORS, FONTS } from './../../ui';

interface Props {
  width?: string;
  fullWidth?: boolean;
  error?: boolean;
  disabled?: boolean;
}

export const TextAreaStyled = styled.textarea`
  width: ${(props: Props) => props.fullWidth ? '100%' : props.width};
  height: 100px;
  font-family: ${FONTS.ROBOTO_REGULAR};
  font-size: 14px;
  resize: none;
  padding: 12px;
  background: ${COLORS.WHITE};
  border: 2px solid ${COLORS.BORDER_GRAY};
  border-radius: 4px;
  outline: 0;

  ${(props: Props) => props.error && `
    border: 2px solid ${COLORS.ERROR_RED};
  `}

  ${(props: Props) => props.disabled && `
    pointer-events: none;
    opacity: 0.3;
  `}

  &:focus{
    border: 2px solid ${COLORS.AQUA_BLUE};
  }

  &::placeholder{
    color: ${COLORS.TEXT_GRAY};
    font-family: ${FONTS.ROBOTO_REGULAR}
  }
`;
