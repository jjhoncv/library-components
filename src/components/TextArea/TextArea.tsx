import * as React from 'react';
import * as PropTypes from 'prop-types';
import { useState } from 'react';
import { TextAreaStyled } from './styled';

interface Props {
  text?: string;
  width?: string;
  fullWidth?: boolean;
  placeholder?: string;
  disabled?: boolean;
  error?: boolean;
  maxLength?: string;
  minLength?: string;
  getText?: Function;
}

export const TextArea: React.FC<Props> = ({
  text,
  getText,
  ...nextProps
}) => {

  const [ value, setValue ] = useState(text);

  const handleChange = (value) => {
    setValue(value);
    getText && getText(value);
  }

  return(
    <TextAreaStyled
      value={value}
      onChange={(e) => handleChange(e.target.value)}
      {...nextProps}
    />
  );
}

TextArea.propTypes = {
  text: PropTypes.string,
  width: PropTypes.string,
  fullWidth: PropTypes.bool,
  placeholder: PropTypes.string,
  disabled: PropTypes.bool,
  error: PropTypes.bool,
  maxLength: PropTypes.string,
  minLength: PropTypes.string,
  getText: PropTypes.func
}

TextArea.defaultProps = {
  width: '420px'
}
