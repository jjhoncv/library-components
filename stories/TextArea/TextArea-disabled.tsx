import * as React from 'react';
import { WithPropTypes } from '../../.storybook/Containers/WithPropTypes';
import { TextArea } from '../../src/components/TextArea';

const Wrapper = () => (
  <TextArea text="Descripcion" disabled/>
);

export default () => {
  return WithPropTypes({
    Component: TextArea,
    RenderComponent: Wrapper,
    showProp: false
  });
}
