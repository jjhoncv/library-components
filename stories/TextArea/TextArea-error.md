# TextArea

Importaci√≥n del componente TextArea.

```js
import { TextArea } from 'library/components/TextArea';
```

<!-- STORY -->

#### Uso

```jsx
<TextArea text="Descripcion" error />
```
