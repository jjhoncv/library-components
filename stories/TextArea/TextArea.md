# TextArea

Importaci√≥n del componente TextArea.

```js
import { TextArea } from 'library/components/TextArea';
```

<!-- STORY -->

#### Uso

```jsx
<TextArea
  placeholder="Ingrese una descripcion"
  getText={(text) => console.log(text)}
/>
```
