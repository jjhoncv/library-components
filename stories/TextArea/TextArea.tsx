import * as React from 'react';
import { WithPropTypes } from '../../.storybook/Containers/WithPropTypes';
import { TextArea } from '../../src/components/TextArea';

const Wrapper = () => (
  <TextArea
    placeholder="Ingrese una descripcion"
    getText={(text) => console.log(text)}
  />
);

export default () => {
  return WithPropTypes({
    Component: TextArea,
    RenderComponent: Wrapper,
    showProp: false
  });
}
