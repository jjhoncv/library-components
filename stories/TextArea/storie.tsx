import { storiesOf } from '@storybook/react';

import TextArea from './TextArea';
import * as readmeTextArea from './TextArea.md';
import TextAreaDisabled from './TextArea-disabled';
import * as readmeTextAreaDisabled from './TextArea-disabled.md';
import TextAreaError from './TextArea-error';
import * as readmeTextAreaError from './TextArea-error.md';
import TextAreaText from './TextArea-text';
import * as readmeTextAreaText from './TextArea-text.md';

const stories = storiesOf('2. Components/TextArea', module);

stories.add('Default', TextArea(), {
  readme: { content: readmeTextArea }
});
stories.add('Disabled', TextAreaDisabled(), {
  readme: { content: readmeTextAreaDisabled }
});
stories.add('With Error', TextAreaError(), {
  readme: { content: readmeTextAreaError }
});
stories.add('With Text', TextAreaText(), {
  readme: { content: readmeTextAreaText }
});
