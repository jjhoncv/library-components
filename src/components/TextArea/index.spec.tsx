import * as React from 'react';
import { shallow } from 'enzyme';
import { TextArea } from './TextArea';

describe('TextArea Component', () => {
  describe('TextArea => render snapshot', () => {
    it('Render correctly', () => {
      const wrapper = shallow(
        <TextArea />
      );
      expect(wrapper.html()).toMatchSnapshot();
    })
  });

  describe('TextArea Builder', () => {
    it('TextArea => Render prop disabled', () => {
      const wrapper = shallow(
        <TextArea disabled/>
      );
      expect(wrapper.prop('disabled')).toBe(true);
    });
    it('TextArea => Render prop error', () => {
      const wrapper = shallow(
        <TextArea error/>
      );
      expect(wrapper.prop('error')).toBe(true);
    });
    it('TextArea => Render value', () => {
      const wrapper = shallow(
        <TextArea text="Descripción"/>
      );
      expect(wrapper.prop('value')).toBe("Descripción");
    });
    it('TextArea => Render prop getText onChange', () => {
      const mockFn = jest.fn();
      const wrapper = shallow(
        <TextArea text="Descripción" getText={mockFn}/>
      );
      wrapper.simulate('change', {target: {value: 'test'}});
      expect(mockFn.mock.calls[0][0]).toBe('test');
    });
  });
});
