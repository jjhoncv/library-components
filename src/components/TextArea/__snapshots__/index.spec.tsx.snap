// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component TextArea => render snapshot Render correctly 1`] = `"<textarea width=\\"420px\\" class=\\"sc-bdVaJa hCnJxf\\"></textarea>"`;
